### DISCUSSION TITLE
Chart in pdf

### DISCUSSION THREAD

**Discussion by RVP04** [spatie/laravel-pdf#181] [Q&A]
I am using chartjs library already in the application. Need to draw the chart in the pdf and tuts or leads will be helpful

**Comment by ArielMejiaDev** [comment]
@RVP04  Here you can find a good example: https://www.youtube.com/watch?v=a6pVPcUiQ9U&t=97s

**Reply by RVP04** [reply]
I tried to generate the graph and which comes complete incorrect. Why can't we generate you the same package. (I know this package uses browershot) 

In view it is correct and only in pdf it is incorrect.

**Reply by RVP04** [reply]
'''

use Spatie\LaravelPdf\Facades\Pdf;

Pdf::view('test')
    ->withBrowsershot(function (Browsershot $browsershot) {
        $browsershot->scale(0.5);
    })
    ->save($this->targetPath);
'''

Whether can I use the browsershot view in this example

**Comment by RVP04** [comment]
I generated the chart but see difference sir 
this is generated in view 
<img width="1427" alt="Screenshot 2024-09-14 at 17 17 40" src="https://github.com/user-attachments/assets/01a55dac-24b6-44c0-acca-1db0d3aedf86">
this is generated in pdf
<img width="1089" alt="Screenshot 2024-09-14 at 17 17 51" src="https://github.com/user-attachments/assets/260693f0-ccab-4510-9cf1-3f65422a8785">
how this much difference will come sir

**Reply by ArielMejiaDev** [reply]
Looks different, probably did you use the same library as the tutorial?

**Reply by RVP04** [reply]
yes
```
$html = view('reports.feedback.pdf.feedback_completed_teacher', [
            'teachers' => $groupedData,
            'branchDepartmentAverages' => $branchDepartmentAverage
        ])->with([
            'reportTitle' => 'Teacher Feedback Report ' . $semester->name,
            'branchName' => $branch ?? ''
        ])->render();

$pdfContent = Browsershot::html($html)
            ->format('A4')
            ->landscape() // Sets orientation to landscape
            ->margins(25, 5, 12, 5, 'mm') // Margins in mm
            ->pdf(); // Save the PDF with the specified name

        // Return the PDF to the browser as a downloadable file
        return Response::make($pdfContent, 200, [
            'Content-Type' => 'application/pdf',
            'Content-Disposition' => 'attachment; filename="Teacher Feedback Report.pdf"',
        ]);
```
same code it is coming correctly in the view only in pdf

**Reply by RVP04** [reply]
He is using highCharts and TailwindCSS. I will look and get back to you sir.

**Reply by RVP04** [reply]
i dont use tailwindcss for rendering, because of that chart is not fully loaded. i dont know what to do

**Reply by RVP04** [reply]
I tried to render the same using tailwindcss but still it generates only some portion not 100 width
<img width="1117" alt="Screenshot 2024-09-14 at 18 25 31" src="https://github.com/user-attachments/assets/688131c9-0796-40e0-91a5-95c97c004e56">

**Reply by RVP04** [reply]
> Looks different, probably did you use the same library as the tutorial?

I have also asked the author by using tailwindcss let's hope for the best reply 

@freekmurze @ArielMejiaDev 

Let also wait for authors 

Hoping for positive reply ... I may need to generate more than 100 charts in different pages
